core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -1207,4 +1213,17 @@ public Object clone() throws CloneNotSupportedException {
     r.colVisibility = Arrays.copyOf(colVisibility, colVisibility.length);
     return r;
   }
+
+  /**
+   * Checks the max size of a key because external operations, such as writing to RFiles, require the key size fits within [0,Integer.MAX_VALUE]
+   *
+   * @return true if the key is a legal size
+   */
+  public boolean sanityCheckKey() {
+    // If fails the check
+    if ((long) ((long) this.row.length + (long) this.colFamily.length + (long) this.colQualifier.length + (long) this.colVisibility.length) > Integer.MAX_VALUE) {
```

Comment:
There's a few extra casts here. You only need to cast the first int to a long in this sum. Java will cast the rest. If you want to keep those, to be more explicit, that's fine, but the outer cast is redundant, as this is already a summation of longs.